src/freenet/clients/http/wizardsteps/BandwidthLimit.java

```diff
@@ -25,4 +39,31 @@ public BandwidthLimit(long downBytes, long upBytes, String descriptionKey, boole
 		this.descriptionKey = descriptionKey;
 		this.maybeDefault = maybeDefault;
 	}
+
+	/**
+	 * Calculate download and upload limit
+	 * @param bytesPerMonth monthly bandwidth limit
+	 */
+	public BandwidthLimit(long bytesPerMonth) {
+		/*
+		 * Fraction of total limit used for download. Asymptotically from 0.5 at the minimum cap to 0.8.
+		 *
+		 * FIXME: Why do we do this? It does not actually work, since
+		 * download cannot be larger than upload for any long amount
+		 * of time.
```

Comment:
While browsing, Freenet uses realtime mode which allows for short-term bursts of much higher download bandwidth usage. People are sometimes using that to boost a specific download (by simply accessing the file interactively).

Reply:
Upload is limited because maxing it out increases latency... http://bufferbloat.net/

And fred (line most layered P2Ps) deals very poorly with high-latency links

Reply:
Can I add this explanation to the comment and change FIXME to Q&A?

Reply:
Sure.